File: spare_parts_shop_flutter/lib/models/login_response.dart
class LoginResponse {
  final int? userId;
  final String? token;
  final String? username;
  final String? role;
  final int? businessId;
  final int? branchId;
  final bool? mustChangePassword;
  final dynamic configuration;
  final String? message;

  LoginResponse({
    this.userId,
    this.token,
    this.username,
    this.role,
    this.businessId,
    this.branchId,
    this.mustChangePassword = false,
    this.configuration,
    this.message,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    print('=== Parsing LoginResponse ===');
    print('JSON: $json');
    return LoginResponse(
      userId: (json['userId'] as num?)?.toInt(),
      token: json['token'] as String?,
      username: json['username'] as String?,
      role: json['role'] as String?,
      businessId: (json['businessId'] as num?)?.toInt(),
      branchId: (json['branchId'] as num?)?.toInt(),
      mustChangePassword: json['mustChangePassword'] as bool? ?? false,
      configuration: json['configuration'],
      message: json['message'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'token': token,
      'username': username,
      'role': role,
      'businessId': businessId,
      'branchId': branchId,
      'mustChangePassword': mustChangePassword,
      'configuration': configuration,
      'message': message,
    };
  }
}

File: spare-parts-shop/src/main/java/com/spareparts/dto/LoginResponse.java
package com.spareparts.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.Map;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class LoginResponse {
    private Long userId;
    private String username;
    private String role;
    private Long businessId;
    private Long branchId;
    private Boolean mustChangePassword = false;
    private BusinessConfigurationDto configuration;
    private java.util.Set<String> permissions;
    private String message;
    private String token; // For backward compatibility

    public LoginResponse(Long userId, String username, String role, Long businessId, Long branchId, BusinessConfigurationDto configuration, String message) {
        this.userId = userId;
        this.username = username;
        this.role = role;
        this.businessId = businessId;
        this.branchId = branchId;
        this.mustChangePassword = false;
        this.configuration = configuration;
        this.permissions = new java.util.HashSet<>();
        this.message = message;
        this.token = null;
    }

    public LoginResponse(Long userId, String username, String role, Long businessId, Long branchId, Boolean mustChangePassword, BusinessConfigurationDto configuration, String message) {
        this.userId = userId;
        this.username = username;
        this.role = role;
        this.businessId = businessId;
        this.branchId = branchId;
        this.mustChangePassword = mustChangePassword;
        this.configuration = configuration;
        this.permissions = new java.util.HashSet<>();
        this.message = message;
        this.token = null;
    }

    public LoginResponse(Long userId, String username, String role, Long businessId, BusinessConfigurationDto configuration, String message) {
        this.userId = userId;
        this.username = username;
        this.role = role;
        this.businessId = businessId;
        this.branchId = null;
        this.mustChangePassword = false;
        this.configuration = configuration;
        this.permissions = new java.util.HashSet<>();
        this.message = message;
        this.token = null;
    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class BusinessConfigurationDto {
        private String businessType;
        private String billingType;
        private String currency;
        private String timezone;
        private String financialYear;
        private List<Map<String, Object>> modules;
        private List<Map<String, Object>> permissions;
        private Map<String, Object> invoiceSettings;
        private Map<String, Object> dashboardSettings;
        private Map<String, Object> notificationSettings;
        private Map<String, Object> theme;
    }
}
